src/components/SpeciesComposition/SpeciesComposition.vue

```diff
@@ -0,0 +1,131 @@
+<template>
+  <v-card-subtitle v-if="message">{{ message }}</v-card-subtitle>
+  <div v-else>
+    <v-chart ref="chart" class="chart" :option="option" />
+    <download-chart-button filename="species-composition.png" />
+  </div>
+</template>
+
+<script>
+import axios from "axios";
+import tabs from "@/data/tabs";
+import { use } from "echarts/core";
+import { CanvasRenderer } from "echarts/renderers";
+import { PieChart } from "echarts/charts";
+import { TooltipComponent } from "echarts/components";
+import VChart from "vue-echarts";
+import DownloadChartButton from "../DownloadChartButton/DownloadChartButton.vue";
+import buildFeatureUrl from "@/lib/build-feature-url";
+
+use([CanvasRenderer, PieChart, TooltipComponent]);
+
+export default {
```

Comment:
So I don't know why we decided to use Vue 2, but I do now realize that we have a lot of repeated logic for the charts using the options API. Maybe if we used the composition API we could reduce some of the logic. So we could either install the @vue/composition-api (which is already a dependency) as a Vue plugin or we could upgrade Vue to version 2.7 which has support for the composition API. I prefer the upgrading since we then could also remove the @vue/composition-api dependency, thus having one less dependency. Btw I'm not saying we should use the composition API for the whole codebase because that would be a big refactor. Just for the chart components would be enough.

Reply:
We are using Vue 2 because `vue2mapbox-gl` didn't work on Vue 3. I can make a ticket for it, but considering the current flow of a week of work and then a break I don't know when it will be a good time to do this.